lib/fluent/log.rb

```diff
@@ -281,6 +284,7 @@ def event(level, args)
       }
 
       map.each_pair {|k,v|
+        next if (k.to_s == "plugin_type" || k.to_s == "plugin_id") and @optional_header
```

Comment:
It's very exceptional case, isn't it?
If you want to remove it even in `RootAgent#shutdown`, you can remove these attributes from `root_agent.rb`.